File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque 

def solution(priorities, location):
    answer = 0
    priorities = deque(priorities)
    index = deque(range(len(priorities)))
    
    while priorities:
        maxP = max(priorities)
        
        if priorities[0] == maxP:
            priorities.popleft()
            answer += 1
            if index.popleft() == location:
                break
        else:
            priorities.append(priorities.popleft())
            index.append(index.popleft())
        
    return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    let index = 0;
    let arr = [];
    let max = Math.max(...priorities);
    let answer = 0;
    
    for(let i = 0; i < priorities.length; i ++) {
        arr.push(i);
    }
    
    while(priorities.length != 0) {
        max = Math.max(...priorities);
        
        if(priorities[0] < max) {
            priorities.push(priorities.shift());
            arr.push(arr.shift());
        } else {
            answer ++;
            priorities.shift();
            
            if(arr.shift() == location) {
                return answer;
            }
        }
        
    }
}